WordPress/AbstractFunctionRestrictionsSniff.php

```diff
@@ -198,27 +224,40 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
 				continue;
 			}
 
-			if ( preg_match( $group['regex'], $token_content ) < 1 ) {
-				continue;
+			if ( preg_match( $group['regex'], $token_content ) === 1 ) {
+				$this->process_matched_token( $stackPtr, $groupName, $token_content );
 			}
+		}
 
-			if ( 'warning' === $group['type'] ) {
-				$addWhat = array( $phpcsFile, 'addWarning' );
-			} else {
-				$addWhat = array( $phpcsFile, 'addError' );
-			}
+	} // End is_targetted_token().
 
-			call_user_func(
-				$addWhat,
-				$group['message'],
-				$stackPtr,
-				$groupName,
-				array( $token_content )
-			);
+	/**
+	 * Process a matched token.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 *
+	 * @return void
+	 */
+	public function process_matched_token( $stackPtr, $group_name, $matched_content ) {
 
+		if ( 'warning' === $this->groups[ $group_name ]['type'] ) {
+			$addWhat = array( $this->phpcsFile, 'addWarning' );
```

Comment:
I think it would be better to directly call the method `addWarning` like is done further done in [#diff-ede7f8be23ca928d9ba5e981dc4c08fbR1137](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/795/files#diff-ede7f8be23ca928d9ba5e981dc4c08fbR1137) instead of using `call_user_func()`